alchemy/src/cloudflare/dispatch-namespace.ts

```diff
@@ -0,0 +1,222 @@
+import {
+  createCloudflareApi,
+  type CloudflareApi,
+  type CloudflareApiOptions,
+} from "./api.ts";
+import type { Bound } from "./bound.ts";
+
+/**
+ * Properties for creating or updating a Dispatch Namespace
+ */
+export interface DispatchNamespaceProps extends CloudflareApiOptions {
+  /**
+   * Name of the namespace
+   */
+  namespace?: string;
+
+  /**
+   * Whether to adopt an existing namespace with the same name if it exists
+   * If true and a namespace with the same name exists, it will be adopted rather than creating a new one
+   *
+   * @default false
+   */
+  adopt?: boolean;
+
+  /**
+   * Whether to delete the namespace.
+   * If set to false, the namespace will remain but the resource will be removed from state
+   *
+   * @default true
+   */
+  delete?: boolean;
+}
+
+export function isDispatchNamespace(
+  resource: Resource,
+): resource is DispatchNamespaceResource {
+  return resource[ResourceKind] === "cloudflare::DispatchNamespace";
+}
+
+/**
+ * Output returned after Dispatch Namespace creation/update
+ */
+export interface DispatchNamespaceResource
+  extends Resource<"cloudflare::DispatchNamespace">,
+    Omit<DispatchNamespaceProps, "delete"> {
+  type: "dispatch_namespace";
+  /**
+   * The name of the namespace
+   */
+  namespace: string;
+
+  /**
+   * Time at which the namespace was created
+   */
+  createdAt: number;
+
+  /**
+   * Time at which the namespace was last modified
+   */
+  modifiedAt: number;
+}
+
+export type DispatchNamespace = DispatchNamespaceResource &
+  Bound<DispatchNamespaceResource>;
+
+/**
+ * A Cloudflare Dispatch Namespace enables routing worker requests to different scripts based on patterns.
+ *
+ * @see https://developers.cloudflare.com/workers/configuration/routing/dispatch-namespace/
+ *
+ * @example
+ * // Create a basic dispatch namespace
+ * const myNamespace = await DispatchNamespace("my-namespace", {
+ *   namespace: "my-namespace"
+ * });
+ *
+ * @example
+ * // Create dispatch namespace with default name from id
+ * const dispatchNS = await DispatchNamespace("api-dispatch");
+ *
+ * @example
+ * // Adopt an existing namespace if it already exists instead of failing
+ * const existingNamespace = await DispatchNamespace("existing-ns", {
+ *   namespace: "existing-namespace",
+ *   adopt: true
+ * });
+ *
+ * @example
+ * // When removing from Alchemy state, keep the namespace in Cloudflare
+ * const preservedNamespace = await DispatchNamespace("preserve-ns", {
+ *   namespace: "preserved-namespace",
+ *   delete: false
+ * });
+ */
+
+export async function DispatchNamespace(
+  name: string,
+  props: DispatchNamespaceProps = {},
+): Promise<DispatchNamespace> {
+  const dispatchNamespace = await _DispatchNamespace(name, props);
+  const binding = await bind(dispatchNamespace);
+  return {
+    ...dispatchNamespace,
+    // Add any runtime methods that dispatch namespaces might have
+    // For now, this is mostly for type compatibility
+    ...binding,
```

Comment:
you can't spread these because it's a proxy - you need to explicitly add each method that is on the runtime binding value. you can look at other resources to understand